oss_src/sframe_query_engine/planning/optimizations/union_transforms.hpp

```diff
@@ -99,10 +104,23 @@ class opt_union_on_source : public opt_union_transform {
           size_t begin_index = n->inputs[i]->p("begin_index");
           size_t end_index   = n->inputs[i]->p("end_index");
 
-          distinct_input_ranges.insert({begin_index, end_index});
+          size_t size = 0;
+
+          if(t == planner_node_type::SFRAME_SOURCE_NODE) {
+            // Get the size of the sframe
+            size = n->inputs[i]->any_p<sframe>("sframe").num_rows();
+          } else {
+            // Get the size of the sarray
```

Comment:
Is it possible `planner_node_type` will ever have values other than sframe and sarray? When traversing enums I like to use:

```
switch (enum_value) {
    case value_1:
        // logic here
        break;
    case value_2:
        // logic here
        break;
}
```

If you omit a default path, you get a compiler warning/error if the enum ever changes to have more values since you'll no longer cover all the cases.